crates/ra_hir/src/code_model.rs

```diff
@@ -1300,6 +1304,38 @@ impl From<PerNs> for ScopeDef {
     }
 }
 
+impl ScopeDef {
+    fn all_items(def: PerNs) -> Vec<Self> {
```

Comment:
Let's either return an `ArrayVec<[ScopeDef; 2]>` here, or maybe just accept an `acc: &mut Vec<ScopeDef>` as an argument, to avoid the needless allocation.

Reply:
I think it can return 3 results: a type, a value and a macro.

Reply:
Edited the comment!

Reply:
Where did you define the function `O` in that snippet? :)

Reply:
I replaced this `ArrayVec<[Self; 3]>`. Let me know if that isn't what you were looking for.